### PR TITLE
Added Cdn & fixed tests accordingly

### PR DESCRIPTION
**Pull request by KennethJakobsen**

### CODE REVIEW

File: src/UmbracoFileSystemProviders.Azure.Tests/FileSystemProviders.upgrade.config [hunk 1]
```diff
@@ -13,6 +13,7 @@
         Defaults to 365 days.
       -->
   		<add key="maxDays" value="365"/>
+      <add key="cdnUrl" value="[myCDN].azureedge.net" />
```
Comment (JimBobSquarePants) [inline src/UmbracoFileSystemProviders.Azure.Tests/FileSystemProviders.upgrade.config]: If no CDN is to be added will this break the configuration?

File: src/UmbracoFileSystemProviders.Azure.Tests/UmbracoFileSystemProviders.Azure.Tests.csproj [hunk 2]
```diff
@@ -276,6 +276,9 @@
     <Analyzer Include="..\packages\StyleCop.Analyzers.1.0.0-beta014\analyzers\dotnet\cs\Newtonsoft.Json.dll" />
     <Analyzer Include="..\packages\StyleCop.Analyzers.1.0.0-beta014\analyzers\dotnet\cs\StyleCop.Analyzers.dll" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (JimBobSquarePants) [inline src/UmbracoFileSystemProviders.Azure.Tests/UmbracoFileSystemProviders.Azure.Tests.csproj]: What's this?

**Comment by JimBobSquarePants** [comment]
Thanks for this, much appreciated. I've added a few questions to your PR just to clarify some things for me. One additional thing I'd like to know is whether setting the url to the CDN affects the ability to upload files. The CDN will be readonly so have you separated the process of saving retrieving media? I couldn't see any code.

File: src/UmbracoFileSystemProviders.Azure.Tests/FileSystemProviders.upgrade.config (hunk 1 above)
Comment (KennethJakobsen) [inline src/UmbracoFileSystemProviders.Azure.Tests/FileSystemProviders.upgrade.config]: No, It is optional

File: src/UmbracoFileSystemProviders.Azure.Tests/UmbracoFileSystemProviders.Azure.Tests.csproj (hunk 2 above)
Comment (KennethJakobsen) [inline src/UmbracoFileSystemProviders.Azure.Tests/UmbracoFileSystemProviders.Azure.Tests.csproj]: I Really don't know, nothing i've added on purpose

File: src/UmbracoFileSystemProviders.Azure.Tests/FileSystemProviders.upgrade.config (hunk 1 above)
Comment (JimBobSquarePants) [inline src/UmbracoFileSystemProviders.Azure.Tests/FileSystemProviders.upgrade.config]: Thanks!

File: src/UmbracoFileSystemProviders.Azure.Tests/UmbracoFileSystemProviders.Azure.Tests.csproj (hunk 2 above)
Comment (JimBobSquarePants) [inline src/UmbracoFileSystemProviders.Azure.Tests/UmbracoFileSystemProviders.Azure.Tests.csproj]: MSBuild at it's finest!

**Comment by KennethJakobsen** [comment]
The azure cdn is an eventually consistent resource, the cdn does not need any code to upload media as it is configured to look in the blob for files not alreadypresent in the cdn.

**Comment by JimBobSquarePants** [comment]
Och I managed to convince myself that `ResolveUrl(string path, bool relative)` was used by the add/delete methods also. Obviously not the case. I'll get this merged asap. Thanks!

**Pull request merged**